plugin/storage/es/spanstore/reader_test.go

```diff
@@ -1267,36 +1252,34 @@ func TestSpanReader_GetEmptyIndex(t *testing.T) {
 }
 
 func TestSpanReader_ArchiveTraces(t *testing.T) {
-	withArchiveSpanReader(t, false, func(r *spanReaderTest) {
-		mockSearchService(r).
-			Return(&elastic.SearchResult{}, nil)
-		mockArchiveMultiSearchService(r, "jaeger-span-archive").
-			Return(&elastic.MultiSearchResult{
-				Responses: []*elastic.SearchResult{},
-			}, nil)
-		query := spanstore.GetTraceParameters{}
-		trace, err := r.reader.GetTrace(context.Background(), query)
-		require.NotEmpty(t, r.traceBuffer.GetSpans(), "Spans recorded")
-		require.Nil(t, trace)
-		require.EqualError(t, err, "trace not found")
-	})
-}
-
-func TestSpanReader_ArchiveTraces_ReadAlias(t *testing.T) {
-	withArchiveSpanReader(t, true, func(r *spanReaderTest) {
-		mockSearchService(r).
-			Return(&elastic.SearchResult{}, nil)
-		mockArchiveMultiSearchService(r, "jaeger-span-archive-read").
-			Return(&elastic.MultiSearchResult{
-				Responses: []*elastic.SearchResult{},
-			}, nil)
+	testCases := []struct {
+		useAliases bool
+		suffix     string
+		expected   string
+	}{
+		{false, "", "jaeger-span-"},
+		{true, "", "jaeger-span-read"},
+		{false, "archive", "jaeger-span-"},
```

Comment:
```suggestion
		{false, "foobar", "jaeger-span-"},
```
Is `"jaeger-span-"` correct?

Reply:
doesn't seem it it would match previous behavior

Reply:
@yurishkuro I ran an equivalent test on main
```go
func TestSpanReader_ArchiveTraces(t *testing.T) {
	testCases := []struct {
		useAliases bool
		archive    bool
		expected   string
	}{
		{false, false, "jaeger-span-"},
		{true, false, "jaeger-span-read"},
		{false, true, "jaeger-span-archive"},
		{true, true, "jaeger-span-archive-read"},
	}

	for _, tc := range testCases {
		t.Run(fmt.Sprintf("useAliases=%v archive=%v", tc.useAliases, tc.archive), func(t *testing.T) {
			withArchiveSpanReader(t, tc.useAliases, tc.archive, func(r *spanReaderTest) {
				mockSearchService(r).
					Return(&elastic.SearchResult{}, nil)
				mockArchiveMultiSearchService(r, tc.expected).
					Return(&elastic.MultiSearchResult{
						Responses: []*elastic.SearchResult{},
					}, nil)
				query := spanstore.GetTraceParameters{}
				trace, err := r.reader.GetTrace(context.Background(), query)
				require.NotEmpty(t, r.traceBuffer.GetSpans(), "Spans recorded")
				require.Nil(t, trace)
				require.EqualError(t, err, "trace not found")
			})
		})
	}
}
```
There is a slight difference because we had to change the reader to accomadate some edge cases - with the most important one being the case where archive storage also has `read_aliases` set to true. However, we moved that check to the factory. So while the behaviour of the reader is slightly different, the overall behaviour should be the same which we see from the integration tests passing.

Reply:
For v2, the user can get the archive behaviour by just setting `use_aliases` to `true`. It is a breaking change because the storage won't be reading/writing to and from the `*-archive` index anymore. But I believe that's something that we said we're okay with according to https://github.com/jaegertracing/jaeger/pull/6519

Reply:
v2 callers are still using create-archive functions. Does it mean they will be getting the same index names as before this change without any changes to the config?